homework12/task2.train_class.py

```diff
@@ -0,0 +1,48 @@
+"""Describe the Train object. The class must contain fields and a method for adding wagons
+(it is necessary to add objects, and instances of the wagon class). Describe the class Wagon together with the train.
+The Wagon must contain a list of passengers and allow adding passengers. In the Wagon can be 10 passengers for example.
+When using the len function on a Wagon, I want to see the number of passengers. When using len on a train,
+I want to see a list of Wagons without a locomotive. Each wagon must have a number.
+When printing a wagon to the console, I want to see the following "[n]" where n is the number of the wagon."""
+
+
+class Train:
+    def __init__(self):
+        self._wagons = []
+
+    def __add__(self, wagon):
+        return self._wagons.append(wagon)
+
+    def __repr__(self):
+        final_str = '<=[HEAD]'
+        self._wagons.sort(key=lambda x: x.get_number())
+
+        for wagon in self._wagons:
+            final_str += f'-{wagon}'
+
+        return final_str
+
+    def __len__(self):
+        return len(self._wagons[1:])
```

Comment:
I don't really understand how I coud return a list of Wagons when using len

Reply:
Just print it before returning. I try to do same and it is impossible, because len should return int. My fault